src/sparql/is_a_loop.sparql

```diff
@@ -0,0 +1,13 @@
+prefix obo: <http://purl.obolibrary.org/obo/> 
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+
+SELECT ?entity ?property ?value
+WHERE {
+  ?entity rdfs:subClassOf ?value .
+  ?value rdfs:subClassOf ?entity .
+  ?entity a owl:Class .
+  ?value a owl:Class .
+  ?entity ?property ?value .
```

Comment:
The "property" does not make sense here. You should bind that variable to rdfs:subClassOf, else this could turn out to be owl:equivalentClass, owl:disjointClass or any other relation that could hold between ?entity and ?value.

Reply:
I had to add this to get the result to show up in robot report.
I didn't have a variable for ?property in my first try (which worked with robot query), but the result didn't get added to the output file from report without having ?entity ?property ?value

Reply:
Yes, but this is only randomly giving you the correct results in your case! the ?property could be _any_ property, not necessarily the rdfs:subClassOf one you are checking.

This is how we deal with that in ROBOT report proper:

http://robot.obolibrary.org/report_queries/missing_superclass

i.e. using

```
 VALUES ?property { rdfs:subClassOf }
?value ?property ?entity .
```

Reply:
I think it is not as bad as that because the is_a loop still has to exist between ?entity and ?value, but additional relationships between ?entity and ?value may also be returned.
Will update query to follow the example now.

Reply:
If you have this (very unlikely) ontology:

```
A subclassOf B
B subclassOf A
A skos:closeMatch B
```

Then your query could return:

```
?entity ?property ?value
A skos:closeMatch B
```

Reply:
Is that ok? is an is_a loop allowed with skos:closeMatch?

Reply:
In this case here, the `loop` is still through subClassOf. The check works, say, but the output would be odd: it would show the closeMatch as property despite it having nothing to do with the loop. 

Every triple in a `WHERE` clause looks for _new_ triples as well - so the resulting table will return both

?entity ?property ?value
A skos:closeMatch B
A rdfs:subClassOf B

because both these triples exist in the ontology. However, in both cases, the loop that was identified is the subclassOf loop.. There will be another SPARQL course in a few weeks, I will make sure to invite you so we can discuss this, as this is of importance to everyone!